R/utilites.R

```diff
@@ -287,3 +287,79 @@ IsQTable <- function(x)
     inherits(x, "qTable") ||
     (!is.null(attr(x, "questions")) && !is.null(attr(x, "name")))
 }
+
+
+#' Check that variables supplied belong to a certain question type.
+#'
+#' Helper function to use in Standard R items which require the user
+#' to select variables of a certain question type / variable set
+#' structure.
+#'
+#' @param variables A list or data frame containing the variables to check.
+#' @param required.type A string indicating the required questiontype value.
+#'        Should be one of: "PickAny", "PickOne", "PickOneMulti", "Number",
+#'        "NumberMulti", "NumberGrid", "PickAnyGrid", "Text", "TextMulti",
+#'        "Ranking", or "Experiment".
+#' @param message.prefix Text to display in the error message before the 
+#'        required question type / structure.
+#' @param message.suffix Text to display at the end of the message.
+#'
+#' @export
+RequireQuestionType <- function(variables, required.type, message.prefix, message.suffix) {
+    found.question.types <- vapply(variables,
+                             FUN = function(v) { return(attr(v, "questiontype"))},
+                             FUN.VALUE = character(1))
```

Comment:
Can make this less verbose with
```suggestion
    found.question.types <- vapply(variables,
                                   FUN = attr,
                                   FUN.VALUE = character(1),
                                   which = "questiontype",
                                   exact = TRUE)
```
The last line forces an exact match